src/Microsoft.Identity.Client/Instance/B2COpenIdConfigurationEndpointManager.cs

```diff
@@ -40,10 +41,17 @@ internal class B2COpenIdConfigurationEndpointManager : IOpenIdConfigurationEndpo
             string userPrincipalName,
             RequestContext requestContext)
         {
-            string defaultEndpoint = string.Format(
-                CultureInfo.InvariantCulture,
-                new Uri(authorityInfo.CanonicalAuthority).AbsoluteUri + Constants.OpenIdConfigurationEndpoint);
-            return Task.FromResult(defaultEndpoint);
+            if (B2CAuthority.IsB2CLoginHost(authorityInfo.Host))
+            {
+                string defaultEndpoint = string.Format(
+                    CultureInfo.InvariantCulture,
+                    new Uri(authorityInfo.CanonicalAuthority).AbsoluteUri + Constants.OpenIdConfigurationEndpoint);
+                return Task.FromResult(defaultEndpoint);
+            }
+
+            throw MsalExceptionFactory.GetClientException(
```

Comment:
There is a test called B2CMicrosoftOnlineCreateAuthority that now consistently fails because of this. What suprises me is that it passed on the PR build or it was not run. Investigating...